src/main/java/org/distributed/systems/chord/actors/Statistics.java

```diff
@@ -0,0 +1,167 @@
+
+    private Config config = getContext().getSystem().settings().config();
+    private String centralNodeAddress;
+    private ActorRef centralNode;
+    public static final int m = 12; // Number of bits in key id's
+    public static final long AMOUNT_OF_KEYS = Math.round(Math.pow(2, m));
+
+    public Statistics() {
+        String centralEntityAddress = config.getString("myapp.centralEntityAddress");
+        String centralEntityAddressPort = config.getString("myapp.centralEntityPort");
+        centralNodeAddress = "akka://ChordNetwork@" + centralEntityAddress + ":" + centralEntityAddressPort + "/user/ChordActor";
+    }
+
+    @Override
+    public void preStart() throws Exception {
+        Timeout timeout = Timeout.create(Duration.ofMillis(ChordStart.STANDARD_TIME_OUT));
+        Future<ActorRef> centralNodeFuture = getContext().actorSelection(centralNodeAddress).resolveOne(timeout);
+        centralNode = (ActorRef) Await.result(centralNodeFuture, timeout.duration());
+
+        String port = config.getString("akka.remote.artery.canonical.port");
+        System.out.println("my generated port:" + port);
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .matchEquals("kill", msg -> {
+                    System.out.println("statistics actor gonna kill");
+                    Timeout timeout = Timeout.create(Duration.ofMillis(ChordStart.STANDARD_TIME_OUT));
+                    long generatedKey = 1;
+
+                    ActorRef nodeToKill = null;
+                    boolean randomKeyIsCentral = true;
+                    while(randomKeyIsCentral){
+//                          generate random key
+                        Random rd = new Random();
+                        generatedKey = Math.floorMod(rd.nextLong(), AMOUNT_OF_KEYS);
+
+                        System.out.println("generated key: " + generatedKey);
+
+//                         find successor of random key
+                        Future<Object> findSuccessorFuture = Patterns.ask(centralNode, new FindSuccessor.Request(generatedKey, 0), timeout);
+                        FindSuccessor.Reply rply = (FindSuccessor.Reply) Await.result(findSuccessorFuture, timeout.duration());
+
+                        System.out.println("returned key: " + rply.id);
+                        System.out.println("returned node: " + rply.succesor.toString());
+                        nodeToKill = rply.succesor;
+                        randomKeyIsCentral = centralNode.equals(nodeToKill);
+                        System.out.println("Generatign new key, we dotn wanna kill the cnertal node");
+                        Thread.sleep(100);
+                    }
+
+
+//                    lets kill the node and see how long it takes to stabilize
+                    nodeToKill.tell("killActor", getSelf());
+
+//                    start timer
+                    long start_time = System.currentTimeMillis();
```

Comment:
I'm a bit doubtful on how robust this measurement will be tbh. Maybe we can discuss it in the group.